src/main/java/sorald/event/models/WarningLocation.java

```diff
@@ -11,25 +12,13 @@
     private final Integer endColumn;
     private final String violationSpecifier;
 
-    public WarningLocation(RuleViolation violation) {
-        this.filePath = violation.getFileName();
+    public WarningLocation(RuleViolation violation, Path projectPath) {
+        this.filePath = projectPath.relativize(Paths.get(violation.getFileName())).toString();
         this.startLine = violation.getStartLine();
         this.endLine = violation.getEndLine();
         this.startColumn = violation.getStartCol();
         this.endColumn = violation.getEndCol();
-
-        this.violationSpecifier =
-                violation.getRuleKey()
-                        + File.pathSeparator
-                        + filePath
-                        + File.pathSeparator
-                        + startLine
-                        + File.pathSeparator
-                        + endLine
-                        + File.pathSeparator
-                        + startColumn
-                        + File.pathSeparator
-                        + endColumn;
+        this.violationSpecifier = violation.relativeSpecifier(projectPath);
```

Comment:
This change wasn't strictly necessary, but it reduces code duplication and also fixes the problem of filepaths in mined specifiers being absolute (they are now relative to the project path).

Reply:
Good. I just thought, logically speaking, we should be able to create a WarningLocation without specifying the project path as specified in the command line. But we can add a new constructor later whenever a new way of initialization of WarningLocation was required.